overlord/configstate/config/transaction.go

```diff
@@ -343,6 +400,117 @@ func commitChange(pristine *json.RawMessage, change interface{}) *json.RawMessag
 	panic(fmt.Errorf("internal error: unexpected configuration type %T", change))
 }
 
+// partOfVirtualConfiguration() return true if the requested key
+// is part of a virtual configuration. E.g.:
+// "true" for requestedKey: "a.virtual.b" when virtualKey is "a.virtual".
```

Comment:
maybe the fact that the example had to be presented this way shows that the argument to the functions should be swapped?

anyway for clarity this probably merits to be exported in export_test.go and have a little bit of tests.

This is also true correctly for requestedKey  "a.virtual" and "a" and not "z"

Because of the case of requestedKey: "a"   vs virtualKey "a.virtual" maybe partOf is not the clearest name, maybe overlapWith instead ?